engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -155,14 +160,43 @@ public void processRecord(final TypedRecord<ProcessInstanceMigrationRecord> comm
           }
         });
 
+    final boolean processInstanceHasEventSubprocess =
+        !sourceProcessDefinition.getProcess().getEventSubprocesses().isEmpty();
+
+    if (processInstanceHasEventSubprocess) {
+      responseWriter.writeRejectionOnCommand(
+          command,
+          RejectionType.INVALID_STATE,
+          ERROR_MESSAGE_EVENT_SUBPROCESS_NOT_SUPPORTED_IN_PROCESS_INSTANCE);
+      rejectionWriter.appendRejection(
+          command,
+          RejectionType.INVALID_STATE,
+          ERROR_MESSAGE_EVENT_SUBPROCESS_NOT_SUPPORTED_IN_PROCESS_INSTANCE);
+    }
```

Comment:
:x: We don't stop the processing of the command here. As a result, we write a rejection and an event for the command. 

We need to add a `return;` to stop here.